src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_DashboardStack/Dashboard_TripSearchScreen/use-trips-query.ts

```diff
@@ -29,199 +27,160 @@ export function useTripsQuery(
-            setPageCursor(undefined);
-            if (!isCancel(e)) {
-              setSearchState('search-empty-result');
-              setErrorType(getAxiosErrorType(e));
-              console.warn(e);
-            }
-          }
-        }
-      })();
+    return {
+      tripPatterns: tripPatternsWithKeys,
+      nextPageCursor:
+        searchTime.option === 'arrival'
+          ? results.trip.previousPageCursor
+          : results.trip.nextPageCursor,
+    };
+  };
 
-      cancelTokenRef.current = cancelTokenSource;
-      setErrorType(undefined);
-      return () => {
-        cancelTokenSource.cancel('Unmounting use trips hook');
-      };
-    },
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [fromLocation, toLocation, searchTime, filtersSelection],
+  const {data, error, fetchNextPage, hasNextPage, isFetching, status} =
+    useInfiniteQuery(
+      ['trips', fromLocation, toLocation, searchTime, filtersSelection],
+      fetchTrips,
+      {
+        getNextPageParam: (lastPage) => lastPage.nextPageCursor,
+        retry: 3,
+      },
+    );
+
+  const tripPatterns = useMemo(
+    () =>
+      data
+        ? filterDuplicateTripPatterns(
+            data.pages.flatMap((page) => page.tripPatterns),
+          )
+        : [],
+    [data],
   );
 
+  // Automatically fetch more pages if the total results are below the target
   useEffect(() => {
-    search();
-  }, [search]);
+    if (data && hasNextPage) {
+      const totalResults = data.pages.flatMap(
+        (page) => page.tripPatterns,
+      ).length;
+      if (
+        totalResults < config_target_initial_hits &&
+        data.pages.length < config_max_performed_searches
+      ) {
+        fetchNextPage();
+      }
+    }
+  }, [
+    data,
+    hasNextPage,
+    fetchNextPage,
+    config_target_initial_hits,
+    config_max_performed_searches,
+  ]);
+
+  const addJourneySearchEntryStable = useCallback(addJourneySearchEntry, [
+    addJourneySearchEntry,
+  ]);
+
+  useEffect(() => {
+    if (!fromLocation || !toLocation) return;
+    try {
+      // Fire and forget add journey search entry if both locations are not from geo
+      if (
+        fromLocation.resultType !== 'geolocation' &&
+        toLocation.resultType !== 'geolocation'
+      ) {
+        (async () =>
+          await addJourneySearchEntryStable([fromLocation, toLocation]))();
+      }
+    } catch (e) {}
+  }, [fromLocation, toLocation, addJourneySearchEntryStable]);
 
-  const loadMore = useCallback(() => {
-    return search(pageCursor, tripPatterns);
-  }, [search, pageCursor, tripPatterns]);
+  useEffect(() => {
+    if (status === 'success' && tripPatterns.length > 0) {
+      analytics.logEvent('Trip search', 'Search performed', {
+        searchTime,
+        filtersSelection: toLoggableFiltersSelection(filtersSelection),
+        numberOfHits: tripPatterns.length,
+      });
+    }
+  }, [status, tripPatterns, analytics, searchTime, filtersSelection]);
```

Comment:
Could maybe use a `useDoOnceWhen` instead of useEffect here, or at least consider it. I used it for a similar case in https://github.com/AtB-AS/mittatb-app/pull/5168, and I think it makes sense for things that are basically `onSuccess` callbacks.